### PR TITLE
#3579 Refactored code to use KeyValuePair instead of KeyValuePairList

### PR DESCRIPTION
**Pull request by Arjun-Sharma1**
#3579

### CODE REVIEW

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/GenericData.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void setPair(int index, String key, long value) {
         pairs.add(index, kvp);
     }
 
-    public void setPair(int index, KeyValuePairList kvps) {
+    public void setPair(int index, KeyValuePair kvps) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/GenericData.java]: I think this one should still be `KeyValuePairList kvps` -- it's just for setting a kvpl at a certain index.

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/GenericData.java [hunk 2]
```diff
@@ -62,11 +62,11 @@ public void addPair(String key, long value) {
         pairs.add(kvp);
     }
 
-    public void addPair(KeyValuePairList kvps) {
+    public void addPair(KeyValuePair kvps) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/GenericData.java]: This one should still be `KeyValuePairList`

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/KeyValueBooleanPair.java [hunk 3]
```diff
@@ -73,6 +80,11 @@ public boolean getBooleanValue() {
         return value;
     }
 
+    @Override
+    public ArrayList<KeyValuePair> getList() {
+        return null;
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/KeyValueBooleanPair.java]: this should throw an UnsupportedOperationException (see other methods in this class).  Please make this change in all applicable classes.

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/LogTraceData.java [hunk 4]
```diff
@@ -176,7 +176,7 @@ public void setFormattedMsg(String s) {
         setPair(18, s);
     }
 
-    public void setExtensions(KeyValuePairList kvps) {
+    public void setExtensions(KeyValuePair kvps) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/LogTraceData.java]: This should still be `KeyValuePairList`

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java [hunk 5]
```diff
@@ -422,33 +422,33 @@ protected static String getLogLevel(ArrayList<Pair> pairs) {
         return loglevel;
     }
 
-    public static void handleExtensions(KeyValuePairList extensions, String extKey, String extValue) {
+    public static void handleExtensions(KeyValuePair extensions, String extKey, String extValue) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java]: This should still be `KeyValuePairList`

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/KeyValuePairList.java [hunk 6]
```diff
@@ -12,46 +12,112 @@
...
+    private final String key;
 
-    public KeyValuePairList(String name) {
-        this.name = name;
+    public KeyValuePairList(String key) {
+        this.key = key;
         keyValuePairs = new ArrayList<KeyValuePair>();
     }
 
     public ArrayList<KeyValuePair> getKeyValuePairs() {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/KeyValuePairList.java]: Should remove this method as it's now redundant.

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/KeyValuePairList.java [hunk 7]
```diff
@@ -12,46 +12,108 @@
...
+    }
+
+    @Override
+    public boolean isBoolean() {
+        return false;
+    }
+
+    @Override
+    public String getStringValue() {
+        throw new UnsupportedOperationException("Cannot call getStringValue method on KeyValueListPair class");
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/KeyValuePairList.java]: KeyValueListPair -> KeyValuePairList (multiple times in this file)

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Pull request merged**